src/Content/MemoryContentStorageHandler.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Kirby\Content;
+
+use Kirby\Cache\MemoryCache;
+use Kirby\Cms\Language;
+use Kirby\Cms\ModelWithContent;
+
+/**
+ * @package   Kirby Content
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class MemoryContentStorageHandler extends ContentStorageHandler
+{
+	/**
+	 * Cache instance, used to store content in memory
+	 */
+	protected MemoryCache $cache;
+
+	/**
+	 * Sets up the cache instance
+	 */
+	public function __construct(protected ModelWithContent $model)
+	{
+		parent::__construct($model);
+		$this->cache = new MemoryCache();
+	}
+
+	/**
+	 * Returns a unique id for a combination
+	 * of the version id, the language code and the model id
+	 */
+	protected function cacheId(VersionId $versionId, Language $language): string
+	{
+		return $versionId->value() . '/' . $language->code() . '/' . $this->model->id();
+	}
+
+	/**
+	 * Deletes an existing version in an idempotent way if it was already deleted
+	 */
+	public function delete(VersionId $versionId, Language $language): void
+	{
+		$this->cache->remove($this->cacheId($versionId, $language));
+	}
+
+	/**
+	 * Checks if a version exists
+	 */
+	public function exists(VersionId $versionId, Language $language): bool
+	{
+		$data = $this->cache->get($this->cacheId($versionId, $language));
+
+		// validate the cached data
+		if (
+			isset($data['fields']) === true &&
+			is_array($data['fields']) === true &&
+			isset($data['modified']) === true &&
+			is_int($data['modified']) === true
+		) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Returns the modification timestamp of a version if it exists
+	 */
+	public function modified(VersionId $versionId, Language $language): int|null
+	{
+		if ($this->exists($versionId, $language) === false) {
+			return null;
+		}
+
+		return $this->cache->get($this->cacheId($versionId, $language))['modified'];
```

Comment:
`Question`: Couldn't we use `$cache->modified()` for this?

Reply:
The problem seems to be that the MemoryCache class does not properly update the modification date or at least I get wrong results from calling `MemoryCache::modified` directly.

Reply:
Hm, I wonder why. `MemoryCache::set()` creates a fresh `Cache\Value` instance each time and the `Value` constructor sets `$this->created` to `time()`. So `$cache->modified()` (being an alias for `$cache->created()`) should return the timestamp when the cache value was created.

If we can get it to work with the cache modified timestamp, we could clean up the handler a lot as it would then only need to store the fields array.

Reply:
I will try again today. Maybe I did something wrong in my tests or the implementation.

Reply:
✅